compiler/qsc_qasm3/src/semantic/ast/const_eval.rs

```diff
@@ -600,9 +701,15 @@ fn cast_to_bitarray(ty: &Type, lit: LiteralKind, dims: &ArrayDimensions) -> Opti
 
     match ty {
         Type::Bool(..) => rewrap_lit!(lit, Bool(val), Bitstring(BigInt::from(val), size)),
+        Type::Angle(..) => rewrap_lit!(lit, Angle(val), {
+            if val.size > size {
+                return None;
+            }
+            Bitstring(val.value.into(), size)
+        }),
```

Comment:
Should the types have already been cast to the same width before this cast gets called?